### PR TITLE
Update Solidity contract

### PR DESCRIPTION
**Pull request by ars9**
Solidity contract is updated to utilize latest supported Solidity version and native Solidity errors.

### CODE REVIEW

File: evm/contracts/HapiCore.sol [hunk 1]
```diff
@@ -498,16 +486,18 @@ contract HapiCore is OwnableUpgradeable {
...
+        }
 
-        require(
-            IERC20(_stake_configuration.token).transferFrom(
-                msg.sender,
-                address(this),
-                amount
-            ),
-            "Insufficient tokens or allowance to stake"
+        bool is_transferred = IERC20(_stake_configuration.token).transferFrom(
```
Comment (Sergei2404) [inline evm/contracts/HapiCore.sol:493]: Could we use here safeTransferFrom?

File: evm/contracts/HapiCore.sol [hunk 2]
```diff
@@ -477,14 +464,15 @@ contract HapiCore is OwnableUpgradeable {
     function activateReporter() external {
         uint128 id = getMyReporterId();
 
-        require(id > 0, "Caller is not a reporter");
+        if (id == 0) {
```
Comment (Sergei2404) [inline evm/contracts/HapiCore.sol:467]: Maybe makes sense to create a function like assertNotZero?

File: evm/contracts/HapiCore.sol [hunk 3]
```diff
@@ -1,7 +1,8 @@
 // SPDX-License-Identifier: UNLICENSED
-pragma solidity ^0.8.9;
+pragma solidity ^0.8.22;
```
Comment (Sergei2404) [inline evm/contracts/HapiCore.sol:2]: Why not 0.8.24?

File: evm/contracts/HapiCore.sol [hunk 4]
```diff
@@ -558,27 +549,29 @@ contract HapiCore is OwnableUpgradeable {
...
+        }
 
         // NOTE: Situation where there's not enough tokens to withdraw should be impossible,
         // as the pool is only formed from the tokens staked by the reporters
-        require(
-            IERC20(_stake_configuration.token).transfer(
-                msg.sender,
-                reporter.stake
-            )
+        bool is_transferred = IERC20(_stake_configuration.token).transfer(
```
Comment (Sergei2404) [inline evm/contracts/HapiCore.sol:568]: safeTransfer?

File: evm/contracts/HapiCore.sol (hunk 1 above)
Comment (ars9) [reply evm/contracts/HapiCore.sol:493]: IERC20 doesn't seem to have it.

File: evm/contracts/HapiCore.sol (hunk 3 above)
Comment (ars9) [reply evm/contracts/HapiCore.sol:2]: It's not recommended by hardhat right now.

File: evm/contracts/HapiCore.sol (hunk 4 above)
Comment (ars9) [reply evm/contracts/HapiCore.sol:568]: Also isn't present in IERC20 (nor in HAPI token, which will be used on mainnet).

File: evm/contracts/HapiCore.sol (hunk 2 above)
Comment (ars9) [reply evm/contracts/HapiCore.sol:467]: I would agree with you if it wouldn't be a one-liner that is simply more efficient than a function call.

**Pull request merged**